frontend/controller/serviceworkers/push.js

```diff
@@ -0,0 +1,136 @@
+import { PUBSUB_INSTANCE } from '@controller/instance-keys.js'
+import { makeNotification } from '@model/notifications/nativeNotification.js'
+import sbp from '@sbp/sbp'
+import setupChelonia from '~/frontend/setupChelonia.js'
+import { NOTIFICATION_TYPE, PUBSUB_RECONNECTION_SUCCEEDED, PUSH_SERVER_ACTION_TYPE, REQUEST_TYPE, createMessage } from '~/shared/pubsub.js'
+
+export default (sbp('sbp/selectors/register', {
+  'push/getSubscriptionOptions': (() => {
+    let cachedVapidInformation
+    return () => {
+      if (
+        cachedVapidInformation &&
+        // Cache the VAPID information for one hour. The server public
+        // information should change very infrequently, if it changes at all.
+        (performance.now() - cachedVapidInformation[0]) < 3600e3
```

Comment:
Claude:

> The cache uses performance.now() which resets when the service worker restarts. This could lead to unnecessary refreshes of the VAPID key. Consider using Date.now() instead for more persistent caching.

Reply:
The entire thing resets when the service worker restarts, `performance` or not. And this code doesn't get called every time the service worker restarts.